util/inspect.go

```diff
@@ -0,0 +1,65 @@
+package util
+
+import (
+	"github.com/open-policy-agent/opa/loader"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+func RetrieveRules(paths []string) ([]string, error) {
+	fileNames, err := findRegoFiles(paths)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var publicIds = []string{}
+	for _, fileName := range fileNames {
+		publicId, err := getPublicIdFromFile(fileName)
+		if err != nil {
+			return []string{}, err
+		}
+		if publicId != "" {
+			publicIds = append(publicIds, publicId)
+		}
+	}
+	return publicIds, nil
+}
+
+func findRegoFiles(paths []string) ([]string, error) {
+	fileNames := []string{}
+
+	result, err := loader.AllRegos(paths)
+	if err != nil {
+		return fileNames, err
+	}
+
+	for _, module := range result.Modules {
+		fileName := filepath.Clean(module.Name)
+		if !strings.Contains(fileName, "_test.rego") {
+			fileNames = append(fileNames, fileName)
+		}
+	}
```

Comment:
Does this filepath not also contain the rule id? Would this be a cleaner alternative to trying to extract the rule from the Rego file?

Reply:
It does, but there is nothing stopping customers from changing the `publicId` while keeping the name of the folder unrelated to it. It depends on how strict we want to be but it seemed very likely to me that this could happen